extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmCdiProcessor.java

```diff
@@ -140,28 +144,46 @@ void generateJpaConfigBean(HibernateOrmRecorder recorder,
                 .scope(Singleton.class)
                 .unremovable()
                 .setRuntimeInit()
-                .supplier(recorder.jpaConfigSupplier(hibernateOrmRuntimeConfig))
-                .destroyer(JPAConfig.Destroyer.class);
-
-        // Add a synthetic dependency from JPAConfig to any datasource/pool,
-        // so that JPAConfig is destroyed before the datasource/pool.
-        // The alternative would be adding an application destruction observer
-        // (@Observes @BeforeDestroyed(ApplicationScoped.class)) to JPAConfig,
-        // but that would force initialization of JPAConfig upon application shutdown,
-        // which may cause cascading failures if the shutdown happened before JPAConfig was initialized.
-        if (capabilities.isPresent(Capability.HIBERNATE_REACTIVE)) {
-            configurator.addInjectionPoint(ParameterizedType.create(DotName.createSimple(Instance.class),
-                    new Type[] { ClassType.create(DotName.createSimple("io.vertx.sqlclient.Pool")) }, null),
-                    AnnotationInstance.builder(Any.class).build());
-        } else {
-            configurator.addInjectionPoint(ParameterizedType.create(DotName.createSimple(Instance.class),
-                    new Type[] { ClassType.create(DotName.createSimple(AgroalDataSource.class)) }, null),
-                    AnnotationInstance.builder(Any.class).build());
-        }
+                .supplier(recorder.jpaConfigSupplier(hibernateOrmRuntimeConfig));
 
         syntheticBeanBuildItemBuildProducer.produce(configurator.done());
     }
 
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    void generateJpaConfigBeanObserver(
+            HibernateOrmRecorder recorder,
+            ObserverRegistrationPhaseBuildItem observerRegistrationPhase,
+            BuildProducer<ObserverRegistrationPhaseBuildItem.ObserverConfiguratorBuildItem> observerConfigurationRegistry) {
+        observerConfigurationRegistry.produce(
+                new ObserverRegistrationPhaseBuildItem.ObserverConfiguratorBuildItem(observerRegistrationPhase.getContext()
+                        .configure()
+                        .beanClass(DotName.createSimple("io.quarkus.hibernate.orm.runtime.JPAConfig"))
+                        .observedType(ShutdownEvent.class)
```

Comment:
I assume we have some sort of guarantee that when `ShutdownEvent` is sent, the datasources are still up and usable?

Reply:
mm.. I suppose we do (reading this comment https://github.com/quarkusio/quarkus/issues/34547#issuecomment-2288096186), and if eventually we change the datasource bean to also observe the shutdown event, we would just need to set the correct priority, right?

Reply:
Makes sense